services/packagist/packagist-dependency-version.tester.js

```diff
@@ -0,0 +1,88 @@
+import { isComposerVersion } from '../test-validators.js'
+import { createServiceTester } from '../tester.js'
+export const t = await createServiceTester()
+
+t.create('gets the package version of symfony 5.2.3 - without query params')
+  .get('/symfony/symfony/v5.2.3.json')
+  .expectBadge({
+    label: 'dependency version',
+    message: 'version requirement not found',
+  })
+
+t.create('gets the package version of symfony - with correct dependency')
+  .get('/symfony/symfony/twig/twig.json')
+  .expectBadge({ label: 'twig/twig', message: isComposerVersion })
+
+t.create(
+  'gets the package version of symfony - with correct dependency and specified version'
+)
+  .get('/symfony/symfony/twig/twig.json?version=v3.2.8')
+  .expectBadge({ label: 'twig/twig', message: isComposerVersion })
+
+t.create(
+  'gets the package version of symfony - with correct dependency, correct version and correct custom server'
+)
+  .get(
+    '/symfony/symfony/twig/twig.json?version=v3.2.8&server=https://packagist.org'
+  )
+  .expectBadge({ label: 'twig/twig', message: isComposerVersion })
+
+t.create(
+  'gets the package version of symfony - with correct dependency and specified version and incorrect custom server'
+)
+  .get(
+    '/symfony/symfony/twig/twig.json?version=v3.2.8&server=https://packagisttttttt.org'
+  )
+  .expectBadge({
+    label: 'dependency version',
+    message: 'inaccessible',
+  })
+
+t.create('gets the package version of symfony - with incorrect dependency name')
+  .get('/symfony/symfony/twig/twiiiiiiig.json?version=v3.2.8')
+  .expectBadge({
+    label: 'dependency version',
+    message: 'version requirement not found',
+  })
+
+t.create('gets the package version of symfony - with incorrect version')
+  .get('/symfony/symfony/twig/twig.json?version=v3.2.80000')
+  .expectBadge({
+    label: 'dependency version',
+    message: 'invalid version',
+  })
+
+t.create(
+  `gets the package version of symfony 5.2.3 - incorrectly with missing part of dependency's name`
+)
+  .get('/symfony/symfony/twig.json')
+  .expectBadge({
+    label: 'dependency version',
+    message: 'version requirement not found',
+  })
+
+t.create(
+  'gets the package version of symfony 5.2.3 - name of the dependency contains hyphen'
+)
+  .get('/symfony/symfony/ext-xml.json?version=v5.2.3')
+  .expectBadge({ label: 'ext-xml', message: isComposerVersion })
```

Comment:
We don't really need to explicitly test `-` is a valid character in a URL at the individual service level. It is conventional that any `:param` allows a hyphen in it.
The interesting thing about this test is that what we're getting is the version of a PHP extension: `twig/twig` (`vendor/package`) is a package, `ext-xml` (no vendor) is a language extension.
Also, version=v5.2.3 is probably noise for this test

Reply:
Modified the test to say:
```js
t.create('gets the package version - dependency does not need the vendor')
  .get('/symfony/symfony/ext-xml.json')
  .expectBadge({ label: 'ext-xml', message: isComposerVersion })
```